v2/internal/resolver/resource_hierarchy.go

```diff
@@ -218,16 +218,6 @@ func (h ResourceHierarchy) fullyQualifiedARMIDImpl(subscriptionID string, origin
 			return "", err
 		}
 
```

Comment:
minor: Consider leaving a comment here explaining that we don't check that the owner matches the subscription on purpose?

Something like:

```suggestion
		// We used to have a check here ensuring that the owner ARM ID matched the referenced credential/secret 
		// ARM ID. That check was removed as with ARM ID-based owners the subscription the resource is being
		// deployed to is pretty obvious (it's directly in the ID) so we felt the chances of mistakenly deploying to 
		// an incorrect subscription with kube-based ownership.
```